// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: api/interface/v1/user.proto

package v1

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
)

// Validate checks the field values on UserListReq with the rules defined in
// the proto definition for this message. If any rules are violated, an error
// is returned.
func (m *UserListReq) Validate() error {
	if m == nil {
		return nil
	}

	return nil
}

// UserListReqValidationError is the validation error returned by
// UserListReq.Validate if the designated constraints aren't met.
type UserListReqValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e UserListReqValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e UserListReqValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e UserListReqValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e UserListReqValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e UserListReqValidationError) ErrorName() string { return "UserListReqValidationError" }

// Error satisfies the builtin error interface
func (e UserListReqValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sUserListReq.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = UserListReqValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = UserListReqValidationError{}

// Validate checks the field values on UserList with the rules defined in the
// proto definition for this message. If any rules are violated, an error is returned.
func (m *UserList) Validate() error {
	if m == nil {
		return nil
	}

	for idx, item := range m.GetItems() {
		_, _ = idx, item

		if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return UserListValidationError{
					field:  fmt.Sprintf("Items[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	return nil
}

// UserListValidationError is the validation error returned by
// UserList.Validate if the designated constraints aren't met.
type UserListValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e UserListValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e UserListValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e UserListValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e UserListValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e UserListValidationError) ErrorName() string { return "UserListValidationError" }

// Error satisfies the builtin error interface
func (e UserListValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sUserList.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = UserListValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = UserListValidationError{}

// Validate checks the field values on UserList_UserListItem with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *UserList_UserListItem) Validate() error {
	if m == nil {
		return nil
	}

	return nil
}

// UserList_UserListItemValidationError is the validation error returned by
// UserList_UserListItem.Validate if the designated constraints aren't met.
type UserList_UserListItemValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e UserList_UserListItemValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e UserList_UserListItemValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e UserList_UserListItemValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e UserList_UserListItemValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e UserList_UserListItemValidationError) ErrorName() string {
	return "UserList_UserListItemValidationError"
}

// Error satisfies the builtin error interface
func (e UserList_UserListItemValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sUserList_UserListItem.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = UserList_UserListItemValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = UserList_UserListItemValidationError{}
